hkpy/hkbase/hkobserverfactory.py

```diff
@@ -0,0 +1,111 @@
+###
+# Copyright (c) 2019-present, IBM Research
+# Licensed under The MIT License [see LICENSE for details]
+###
+
+import inspect
+import logging
+import os
+from os.path import dirname, isfile, join
+import traceback
+import glob
+import requests
+import importlib.machinery
+
+from hkpy.hkbase.observer.clients.observerclient import HKBase
+
+clients_by_key = {}
+clients_dirpath = join(dirname(__file__), 'observer', 'clients')
+modules_filepaths = glob.glob(join(clients_dirpath, "*.py"))
+for module_filepath in modules_filepaths:
+    if isfile(module_filepath) and not module_filepath.endswith('__init__.py'):
+        module_name = f"{module_filepath.replace(clients_dirpath, '').replace('.py', '').replace('/', '')}"
+        loader = importlib.machinery.SourceFileLoader(module_name, module_filepath)
+        module = loader.load_module(module_name)
+        for class_name, klass in inspect.getmembers(module, inspect.isclass):
+            if hasattr(klass, 'TYPE_KEY'):
+                clients_by_key[klass.TYPE_KEY] = klass
+
+
+def create_observer(hkbase: HKBase, observer_options=None, hkbase_options=None):
```

Comment:
FYI: Once we migrate to Python 3.7 we should replace this by a TypedDict.

Reply:
observer options?

Reply:
Yes. Annotating the types is easier to maintain than documenting the options in the method's doc string.